### PR TITLE
KFSPTS-29464 Improve ISO 20022 blank country handling

### PR DESCRIPTION
**Pull request by cah292**
This PR improves the country handling in the ISO 20022 extraction. If a transaction has a blank country code but still appears to reference a domestic state code and zip code, then the "US" country code will be used in the file entry. Otherwise, the "ZZ" (Unknown) code will be used instead.

Note that, similar to KualiCo's `PostalCodeValidationServiceImpl` class, the updated code is using a `ZipcodeValidationPattern` instance to assist with the zip code pattern checking.

### CODE REVIEW

File: src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java [hunk 1]
```diff
@@ -833,8 +845,15 @@ private PostalAddress6 constructPostalAddress(
...
+        } else if (stateAndPostalCodeAreWithinUnitedStates(templateCustomerProfile.getStateCode(),
+                templateCustomerProfile.getZipCode())) {
+            LOG.warn("constructPostalAddress, Customer Profile {} has a blank country code but seems to reference "
+                    + "a domestic address; defaulting to '{}'",
+                    templateCustomerProfile.getId(), KFSConstants.COUNTRY_CODE_UNITED_STATES);
+            isoCountryCode = KFSConstants.COUNTRY_CODE_UNITED_STATES;
         } else {
-            LOG.warn("constructPostalAddress, Customer Profile {} has a blank country code; defaulting to '{}'",
+            LOG.warn("constructPostalAddress, Customer Profile {} has a blank country code but seems to reference "
+                    + "a foreign address; defaulting to '{}'",
```
Comment (ajd299) [inline src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java:856]: Suggestion: "defaulting to" => "defaulting iso country code to"

File: src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java [hunk 2]
```diff
@@ -879,8 +898,15 @@ private PostalAddress6 constructPostalAddress(
             isoCountryCode = wasPaymentGroupCreatedPriorToISOCountryConversion(templatePaymentGroup)
                     ? convertFIPSCountryValueToISOCountryCode(countryValue)
                     : convertISOCountryValueToISOCountryCode(countryValue);
+        }  else if (stateAndPostalCodeAreWithinUnitedStates(templatePaymentGroup.getState(),
```
Comment (ajd299) [inline src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java:901]: is there an extra space here (before "else if")?

File: src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java [hunk 3]
```diff
@@ -1027,6 +1053,27 @@ private String convertFIPSCountryNameToISOCountryCode(String fipsCountryName) {
         return isoCountryCode;
     }
 
+    private boolean stateAndPostalCodeAreWithinUnitedStates(String stateCode, String postalCode) {
```
Comment (ajd299) [inline src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java:1056]: I think this method is well written and easy to understand.

File: src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java [hunk 4]
```diff
@@ -833,8 +845,15 @@ private PostalAddress6 constructPostalAddress(
         String isoCountryCode;
         if (StringUtils.isNotBlank(countryCode)) {
             isoCountryCode = convertFIPSCountryCodeToISOCountryCode(countryCode);
+        } else if (stateAndPostalCodeAreWithinUnitedStates(templateCustomerProfile.getStateCode(),
+                templateCustomerProfile.getZipCode())) {
+            LOG.warn("constructPostalAddress, Customer Profile {} has a blank country code but seems to reference "
+                    + "a domestic address; defaulting to '{}'",
+                    templateCustomerProfile.getId(), KFSConstants.COUNTRY_CODE_UNITED_STATES);
+            isoCountryCode = KFSConstants.COUNTRY_CODE_UNITED_STATES;
```
Comment (ajd299) [inline src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java:853]: I think extracting this code to a method would add complexity to the log statement so I am okay with leaving it as is.

**Review by ajd299 — CHANGES_REQUESTED** [review]
Please see my comments

File: src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java (hunk 1 above)
Comment (cah292) [reply src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java:856]: I'll update the wording accordingly. I think the phrase "defaulting to ISO country code" would work a little better; please let me know if you would prefer your suggested wording instead.

File: src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java (hunk 2 above)
Comment (cah292) [reply src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java:901]: Good catch, I'll fix that.

**Comment by cah292** [comment]
I believe I've addressed all of the appropriate follow-up items. This is ready for review again.

**Review by ajd299 — APPROVED** [review]
Changes look good, merging

**Pull request merged**